transaction_data.csv
- SRC :  https://www.kaggle.com/datasets/yogeshtekawade/banking-and-customer-transaction-data
    - customer_data.csv - données démographiques et profils client
    - transaction_data.csv - historique des transactions et investissements
            - Utilité : prédire des événements comme le churn ou la fraude
            - permet de modéliser les comportements financiers individuels, d’identifier des tendances d’investissement ou de dépenses.
    - bank_data.csv - performance financière des agences bancaires données synthétiques mais réalistes, avec du bruit et des valeurs manquantes pour simuler un environnement réel

## Chargement des données
## Nettoyage des données

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt
import os

c:\Users\Admin\anaconda3\Lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


In [2]:
directory_path = r'C:\Users\Admin\Desktop\BankingMLProject\data'
filename = 'transaction_data.csv'
filename_path = os.path.join(directory_path, filename)
print(os.path.join(directory_path, filename))
filename_path

C:\Users\Admin\Desktop\BankingMLProject\data\transaction_data.csv


'C:\\Users\\Admin\\Desktop\\BankingMLProject\\data\\transaction_data.csv'

In [3]:
# Chargement du fichier
df = pd.read_csv(filename_path)

In [4]:
# Affichage des 5 premières lignes pour inspection
df.head()

Transaction_ID  Customer_ID Account_Type  Total_Balance  \
0          300000       209689     Business          69339   
1          300001       206124     Business          12825   
2          300002       207501      Current          67753   
3          300003       208675      Savings          67061   
4          300004       204923     Business           8566   

   Transaction_Amount  Investment_Amount Investment_Type Transaction_Date  
0              4794.0              42580   Fixed Deposit       2024-12-08  
1              3500.0              46605   Fixed Deposit       2022-09-09  
2              2401.0              17027   Fixed Deposit       2022-04-28  
3              2952.0               3054   Fixed Deposit       2023-05-22  
4              1025.0              44937   Fixed Deposit       2023-04-05

In [5]:
print(df.columns)

Index(['Transaction_ID', 'Customer_ID', 'Account_Type', 'Total_Balance',
       'Transaction_Amount', 'Investment_Amount', 'Investment_Type',
       'Transaction_Date'],
      dtype='object')


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Transaction_ID      10000 non-null  int64  
 1   Customer_ID         10000 non-null  int64  
 2   Account_Type        10000 non-null  object 
 3   Total_Balance       10000 non-null  int64  
 4   Transaction_Amount  10000 non-null  float64
 5   Investment_Amount   10000 non-null  int64  
 6   Investment_Type     10000 non-null  object 
 7   Transaction_Date    10000 non-null  object 
dtypes: float64(1), int64(4), object(3)
memory usage: 625.1+ KB


 - conversion nécéssaire des dtypes: float64(1), object(3)
    -  Account_Type        object 
    -  Transaction_Amount  float64 
    -  Investment_Type     object 
    -  Transaction_Date    object

In [7]:
df.describe()

Transaction_ID    Customer_ID  Total_Balance  Transaction_Amount  \
count     10000.00000   10000.000000   10000.000000        10000.000000   
mean     304999.50000  205028.182600   50221.505800         2542.708056   
std        2886.89568    2909.008139   28540.392011         1432.677420   
min      300000.00000  200000.000000    1003.000000           51.613952   
25%      302499.75000  202498.750000   25132.000000         1310.000000   
50%      304999.50000  205060.000000   50326.000000         2538.500000   
75%      307499.25000  207561.000000   74914.500000         3746.250000   
max      309999.00000  209999.000000   99993.000000         7046.296441   

       Investment_Amount  
count       10000.000000  
mean        25550.248400  
std         14108.052078  
min          1001.000000  
25%         13381.000000  
50%         25708.500000  
75%         37731.000000  
max         49998.000000

In [8]:
df.nunique()

Transaction_ID        10000
Customer_ID            6335
Account_Type              3
Total_Balance          9517
Transaction_Amount     4657
Investment_Amount      9029
Investment_Type           3
Transaction_Date       1096
dtype: int64

#Doublons à nettoyer ?
- Transaction_ID        10000
- Customer_ID            6335

## Nettoyage des données

In [9]:
# Vérification des valeurs manquantes
df.isnull().sum()

Transaction_ID        0
Customer_ID           0
Account_Type          0
Total_Balance         0
Transaction_Amount    0
Investment_Amount     0
Investment_Type       0
Transaction_Date      0
dtype: int64

In [10]:
# Vérification des doublons
df.drop_duplicates(inplace=True)


In [11]:
# Encodage des variables catégorielles
df_encoded = pd.get_dummies(df, drop_first=True)

In [12]:
# Sauvegarde du dossier 'transaction_data_cleaned' pour la suite du projet
output_directory = r"C:\Users\Admin\Desktop\BankingMLProject\data"

# Nom du fichier CSV de sortie
output_filename = "transaction_data_cleaned.csv"

# Création du chemin complet du fichier de sortie
output_filepath = os.path.join(output_directory, output_filename)

#Sauvegarder du DataFrame nettoyé en CSV
try:
    # index=False pour ne pas écrire l'index du DataFrame comme une colonne dans le CSV
    df.to_csv(output_filepath, index=False)
    print(f"\nDataFrame nettoyé sauvegardé avec succès dans : {output_filepath}")
except Exception as e:
    print(f"\nUne erreur est survenue lors de la sauvegarde : {e}")


DataFrame nettoyé sauvegardé avec succès dans : C:\Users\Admin\Desktop\BankingMLProject\data\transaction_data_cleaned.csv
